spock-core/src/main/java/org/spockframework/runtime/PlatformSpecRunner.java

```diff
@@ -384,15 +376,26 @@ protected void invoke(SpockExecutionContext context, Object target, MethodInfo m
     if (method == null || method.isExcluded()) {
       return;
     }
+
+    Object[] methodArguments;
+    int parameterCount = method.getReflection() == null ? arguments.length : method.getReflection().getParameterCount();
```

Comment:
in which cases is it possible to for `method.getReflection()`  return `null` ? If that is possible, then you'll have to handle it when accessing the method parameters as I sugested in `MethodInfo`

Reply:
Actually I didn't investigate whihc case it is, but I got a NPE during my tests without the `null`-check.

Reply:
All that call the `org.spockframework.runtime.model.MethodInfo#MethodInfo(org.spockframework.runtime.model.Invoker)` constructor do not have a `reflection` set.
Those are the 9 that are created in `org.spockframework.runtime.PlatformSpecRunner#createMethodInfoForDoRun...`.
They are synthetic method infos that then delegate to the actual methods but they have no real method as backing.
But in those cases, the method parameter accessing is not a problem, because the accessing is only done if at least one of the arguments is matching `MISSING_ARGUMENT` and that can never happen for those.